Classes/View/PreviewView.php

```diff
@@ -503,16 +503,25 @@ protected function getCookie()
      */
     protected function getRecords(PageLayoutView $view, array $row, $area)
     {
+        $pidCondition = sprintf(' AND pid = %d ', (integer) $row['pid']);
+        $workspaceCondition = '';
+        $workspaceId = $this->getActiveWorkspaceId();
+        if ($workspaceId) {
+            $workspaceCondition = sprintf(' AND (t3ver_wsid = %d OR t3ver_wsid = 0)', $workspaceId);
+            $pidCondition = sprintf(' AND (pid = %d OR (pid = -1 AND t3ver_wsid > 0)) ', (integer) $row['pid']);
+        }
```

Comment:
There is, but the ones I found unfortunately only generate the correct conditions for use in frontend. It is possible I missed something though! In any case: the above is approximately the same as those core methods would generate, with the exception of an unwanted "pid" restriction.